### DISCUSSION TITLE
Use fan  entity for dmaker.airfresh.t2017 and dmaker.airfresh.a1

### DISCUSSION THREAD

**Discussion by Kirmas** [XiaoMi/ha_xiaomi_home#494] [Ideas]
Naw for this two airfresh devices you use switch and select entities. My idea is using fan entity lilke Xiomi Milo integration do.
How you do:
![image](https://github.com/user-attachments/assets/299ce2c3-d84f-4d07-a392-3708791ee19a)
My idea is:
![image](https://github.com/user-attachments/assets/8d089bc4-f35d-4b26-a39a-b39ba5592a73)
![image](https://github.com/user-attachments/assets/db60bbd6-db6a-4261-a17c-a7c3792dd95f)

**Comment by SusanPhevos** [comment]
xiaomi_home converts an air-fresh device to a fan entity only if the device's "air-fresh" service has "on" and "fan-level" property both.
[dmaker.airfresh.t2017](http://poc.miot-spec.srv/miot-spec-v2/instance?type=urn:miot-spec-v2:device:air-fresh:0000A012:dmaker-t2017:2) and [dmaker.airfresh.a1](http://poc.miot-spec.srv/miot-spec-v2/instance?type=urn:miot-spec-v2:device:air-fresh:0000A012:dmaker-a1:3) siid=2 air-fresh service only has on property (piid=1). The fan-level property is under siid=7 air-fresh-favorite service. So the [conversion rule](https://github.com/XiaoMi/ha_xiaomi_home/blob/main/custom_components/xiaomi_home/miot/specs/specv2entity.py#L463) is not met.
